### PR TITLE
Add Sandbox project to readme

### PR DESCRIPTION
**Pull request by hakolao**
I've been working on a [2D Pixel Physics Simulator](https://github.com/hakolao/sandbox) using Vulkano for the rendering and compute shaders for CA simulation. The project is more like a tech demo / simulator, but I'll continue on it separately to make a game. But I wanted to release the simulator as is and thought it would be nice to addition to Vulkano showcase. I also use my [egui_winit_vulkano](https://github.com/hakolao/egui_winit_vulkano) integration so I can use [Egui](https://github.com/emilk/egui) for the user interface.

image

Perhaps the weirdest part of this Simulator is just having been able to make the Cellular Automata work with compute shaders.

I sometimes wonder why Vulkano isn't more popular, or if it is, why isn't it visible like `wgpu` is or others. I've found the API much nicer to work with than any other rendering APIs that I've use (limited use). Perhaps I follow wrong places (Reddit and some gamedev stuff mostly).

[Here's a little badly compressed video of it](https://www.youtube.com/watch?v=7B-BkzuzOuk)

Some things that could be cool if resources existed for it:
- How to use Renderdoc with Vulkano (perhaps just how to use renderdoc works, but I haven't yet actually attempted)
- How to do multi-threaded rendering with Vulkano (Will dive into it at some point...)
- Gui with Vulkano (Examples section)

I might take a moment at some point and extract parts from this project to add to the Vulkano examples, for example: "Software Renderer (how to access just pixels...)" or "Drawing on a Canvas", "more complex compute shaders" type of examples... Just because I'd like to help make Vulkano more accessible.

Also I certainly would not mind to get some feedback from you guys!

### CODE REVIEW

**Comment by Rua** [comment]
If you want to help out further to make Vulkano more visible and popular, would you be interested in taking on the maintenance of the website, https://github.com/vulkano-rs/vulkano-www ? It's currently horribly out of date, and with many breaking changes over the last year, most of the code no longer works. This would probably turn off a lot of people.

**Comment by hakolao** [comment]
Yes, I would be interested in that! I agree that updating it would be beneficial for the visibility.

**Comment by Eliah-Lakhin** [comment]
@hakolao You can just fork it and Pull Request back, but please ping me directly to review(I can miss it).

Also, moving further, I think maybe it would be worth to re-implement the Website with tutorial from scratch in this main Vulkano repository(https://github.com/vulkano-rs/vulkano/). Having everything in a single repo would improve maintainability. In particular we can enforce contributors not only upading Documentation and Examples, but also the Website tutorial. And it will be easier for everyone to do everything in a single PR when someone implements braking changes too.

@Rua, @AustinJ235 What do you think about this idea, guys?

**Comment by Eliah-Lakhin** [comment]
The only technical issue with Website is that no one but Pierre has access to publish the current website updates. So even if I merge a PR into vulkano-www I will have to ask him to publish it. I think it could be easier automated if we use GitHub itself as a static website hosting. GitHub works seamlessly in this sense such that if something in the repo it reflects on the websites almost immediately.

I would talk to Pierre about this idea but I think it would be better if by that time we already would have a working website prototype for replacement.

To sum up, we need someone in volunteering to take a lead in implementing a new website prototype. Once I see a Pull Request into this repo with the website I can initiate a conversation with Pierre regarding the domain reconfiguration.

**Comment by Eliah-Lakhin** [comment]
@hakolao By the way, the PR merged. Now we have another full-featured working example in collection of projects using Vulkano, and it looks amazing! It also probably the first example that shows how to integrate egui and Vulkano. Thank you so much for your help! :)

**Comment by hakolao** [comment]
My pleasure @Eliah-Lakhin

**Comment by Eliah-Lakhin** [comment]
> I sometimes wonder why Vulkano isn't more popular, or if it is, why isn't it visible like wgpu is or others.

@hakolao I have found Vulkano API quite user-friendly too. That's why I use it in my personal(yet not published) project for almost 2 years. And IMO Vulkano became even better for the recent years thanks to the community!

The reasons its not enough visible is, yes, first of all the issues you covered: We need better Tutorials, Books, How-to guides. At least we need to update our website :) And maybe now is a good moment to work it out as the API has somewhat stabilized after the series of refactorings made in previous recent months.

Another big thing is that we need to do more highlights in Rust's media. Aside from Reddit there are several sources of information for the community:

1. Rust GameDev monthly digest: https://gamedev.rs/news/029/ It collects all monthly updates from users every month.
2. Rust GameDev monthly meetup coordinated by a former Veloren developer Forest Anderson: https://discord.com/invite/yNtPTb2 By the way, Okko, I think it could be a good place for you to attend and demo your game!
3. Maybe also Rust graphics meetup coordinated by Dzmitry Malyshau a main developer of WGPU, but they are irregular: https://github.com/gfx-rs/meetup

The local Rust communities could be a good place to highlight Vulkano too. For example there is a Russian-speaking video games and graphics developers Rust community(of more than 650 members). I from time to time attending in the chat and on weekly meetups with Vulkano and my personal project that I develop with Vulkano. Btw, I just posted a message about your game to our community in the chat, and the audience was very impressed! I'm going to show it in the next meetup too! :)

**Comment by hakolao** [comment]
Amazing! Yes, I agree with all those points.

I’ll look into the rust gamedev monthly meetup once I get back to work on Monday. That would certainly good to be part of.

**Comment by AustinJ235** [comment]
> The only technical issue with the Website is that no one but Pierre has access to publish website updates. So even if I merge a PR into vulkano-www I will have to ask him to publish it. I think it could be easier automated if we use GitHub itself as a static website hosting. GitHub works seamlessly in this sense such that if something already in the repo, it reflects on the websites almost immediately.
> 
> I would talk to Pierre about this idea but I think it would be better if by that time we already would have a working website prototype for replacement.
> 
> To sum up, we need someone in volunteering to take a lead in implementing a new website prototype. Once I see a Pull Request into this repo with the website I can initiate a conversation with Pierre regarding the domain reconfiguration.

From my understanding anything that goes to master and has a successful build gets published immediately to the website. So in that regard we can continue to update the website. The website itself uses markdown and isn't terribly difficult to modify and I think the system is robust enough. We could probably get away with replacing the web server with another one if we need to.

As for where it is hosted, and where it is tested I am not sure of, but I do know modifications to master do get published automatically.

**Comment by Eliah-Lakhin** [comment]
@AustinJ235 Thank you for clarification! I didn't realize about the auto-publishing. Then it simplifies things.
